osquery/tables/system/windows/registry.cpp

```diff
@@ -116,114 +119,111 @@ inline void explodeRegistryPath(const std::string& path,
 }
 
 /// Microsoft helper function for getting the contents of a registry key
-void queryKey(const std::string& keyPath, QueryData& results) {
+Status queryKey(const std::string& keyPath, QueryData& results) {
   std::string hive;
   std::string key;
   explodeRegistryPath(keyPath, hive, key);
 
   if (kRegistryHives.count(hive) != 1) {
-    return;
+    return Status();
   }
 
-  HKEY hRegistryHandle;
-  auto ret = RegOpenKeyEx(kRegistryHives.at(hive),
-                          TEXT(key.c_str()),
-                          0,
-                          KEY_READ,
-                          &hRegistryHandle);
+  HKEY hkey;
+  auto ret = RegOpenKeyEx(
+      kRegistryHives.at(hive), TEXT(key.c_str()), 0, KEY_READ, &hkey);
+  reg_handle_t hRegistryHandle(hkey, closeRegHandle);
 
   if (ret != ERROR_SUCCESS) {
-    return;
+    return Status(GetLastError(), "Failed to open registry handle");
```

Comment:
I definitely wouldn't want to lose the GLE value, so if using it as the `Status` code is what we're digging then I'm all for it!